diff -Naru a/coders/bmp.c b/coders/bmp.c
--- a/coders/bmp.c	2024-05-25 21:55:36.000000000 +0930
+++ b/coders/bmp.c	2024-06-04 00:15:34.022710478 +0930
@@ -706,9 +706,11 @@
     */
     ignore_filesize=IsStringTrue(GetImageOption(image_info,
       "bmp:ignore-filesize"));
+#if 0
     if ((ignore_filesize == MagickFalse) && (bmp_info.file_size != 0) &&
         ((MagickSizeType) bmp_info.file_size > GetBlobSize(image)))
       ThrowReaderException(CorruptImageError,"ImproperImageHeader");
+#endif
     if (bmp_info.offset_bits < bmp_info.size)
       ThrowReaderException(CorruptImageError,"ImproperImageHeader");
     bmp_info.offset_bits=MagickMax(14+bmp_info.size,bmp_info.offset_bits);
